src/quill-cursors/cursor.ts

```diff
@@ -83,13 +83,16 @@ export default class Cursor {
     setTimeout(() => this._flagEl.classList.remove(Cursor.NO_DELAY_CLASS), this._hideSpeedMs);
   }
 
-  public updateCaret(rectangle: ClientRect): void {
+  public updateCaret(rectangle: ClientRect, container: ClientRect): void {
+    const flagRect = this._flagEl.getBoundingClientRect();
+
     this._caretEl.style.top = `${rectangle.top}px`;
     this._caretEl.style.left = `${rectangle.left}px`;
     this._caretEl.style.height = `${rectangle.height}px`;
 
+    const flagLeft = Math.min(container.width - flagRect.width, rectangle.left);
```

Comment:
I think I was expecting the flag to completely flip horizontally, rather than just get shunted. This looks a bit broken to me:

<img width="161" alt="Screen Shot 2020-01-30 at 13 59 59" src="https://user-images.githubusercontent.com/12036746/73456001-00119200-4369-11ea-9d7b-9dff9c70ee81.png">

Reply:
Hmmm, I see, you want it to stay "flag-ish". 
I have a bit different styles, so this behaviour works better for me. In this way, I'm avoiding "flag" jumping left and right when cursor is getting close to the border, which looks not good when flag is always visible.
Indeed we can make it flippable as you propose, but then it will make things harder for those who want to customise appearance via css or via a custom template.  What do you think?

Reply:
For context, can you please share a screenshot / video of the behaviour you're aiming for with your styles?

Reply:
I think the general solution should probably be:

- add an optional `positionFlag` property to the options: `positionFlag: (flag: HTMLElement, container: HTMLElement) => void`
- if this property is provided, then we call it in `Cursor.updateCaret`
- if this property is _not_ provided, then we run default behaviour of checking for a collision, and just flipping the flag to the left

Reply:
> For context, can you please share a screenshot / video of the behaviour you're aiming for with your styles?

This is my current appearance/behaviour. 

![ezgif com-crop](https://user-images.githubusercontent.com/1318410/73540877-534f1780-4431-11ea-89df-3c2bb0cb7096.gif)

Reply:
Okay thanks. Yes, I think we should flip by default, but allow consumers to override the behaviour by exposing a `positionFlag` method in the options (see above).